src/Microsoft.AspNet.Mvc.ModelBinding/Validation/ModelValidationNode.cs

```diff
@@ -160,7 +160,8 @@ private void ValidateProperties(ModelValidationContext validationContext)
                 // Only want to add errors to ModelState if something doesn't already exist for the property node,
                 // else we could end up with duplicate or irrelevant error messages.
                 var propertyKeyRoot = ModelBindingHelper.CreatePropertyModelName(ModelStateKey,
-                                                                                 propertyMetadata.PropertyName);
+                                                                                 propertyMetadata.BinderModelName
+                                                                                 ?? propertyMetadata.PropertyName);
```

Comment:
nit: rewrap to keep null coalescing expression on one line and don't start a line with a dangling operator:
``` c#
  ...CreatePropertyModelName(
    ModelStateKey,
    propertyMetadata.BinderModelName ?? propertyMetadata.PropertyName);
```